### PR TITLE
Rework `WalletTest::get_confirmed_sends` trait method

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: zcash_client_backend/src/data_api.rs [hunk 1]
```diff
@@ -1270,6 +1272,32 @@ pub trait WalletTest: InputSource + WalletRead {
...
 
+/// The output of a transaction sent by the wallet.
+///
+/// This type is opaque, and exists for use by tests defined in this crate.
+#[cfg(any(test, feature = "test-dependencies"))]
+#[derive(Clone, Debug)]
+pub struct OutputOfSentTx {
+    value: NonNegativeAmount,
+    external_recipient: Option<Address>,
+    ephemeral_address: Option<(Address, u32)>,
```
Comment (nuttycom) [inline zcash_client_backend/src/data_api.rs:1283]: What's the u32 here?

**Review by nuttycom — APPROVED** [review]
utACK with request for additional documentation

File: zcash_client_backend/src/data_api.rs (hunk 1 above)
Comment (str4d) [reply zcash_client_backend/src/data_api.rs:1283]: The index of the ephemeral address.

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/librustzcash/compare/63e80ec630eed9b13be0aef61eb63ac94e6c712f..9b8094b94bf51cf2eba2a6e50a2805ec45477a12) to add requested documentation.

**Review by nuttycom — APPROVED** [review]
utACK 9b8094b94bf51cf2eba2a6e50a2805ec45477a12

**Pull request merged**